build.rs

```diff
@@ -0,0 +1,214 @@
+                    err, account_type
+                ));
+            }
+            if account_type.decimals.len() != account_type.symbols.len() {
+                return Err(format!(
+                    "decimals must be specified for each symbol: {:?}",
+                    account_type
+                ));
+            }
+            if let Some(sig_type) = &account_type.standard_account {
+                match sig_type.as_str() {
+                    "Sr25519"| "Ed25519" | "secp256k1" | "*25519" => {},
+                    _ => {
+                        return Err(format!("Unknown sig type in standardAccount: expected one of Sr25519, Ed25519, secp256k1, *25519: {:?}", account_type))
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+struct AccountType {
+    prefix: u16,
+    network: String,
+    display_name: String,
+    /// If standard account is None then the network is reserved.
+    standard_account: Option<String>,
+    symbols: Vec<String>,
+    decimals: Vec<u8>,
+    website: Option<String>,
+}
+
+impl AccountType {
+    fn name(&self) -> String {
+        format!(
+            "{}Account",
+            inflector::cases::pascalcase::to_pascal_case(&self.network)
+        )
+    }
+
+    fn is_reserved(&self) -> bool {
+        self.standard_account.is_none()
+    }
+}
+
+fn create_ss58_registry(json: &str) -> Result<proc_macro2::TokenStream, String> {
+    let registry: Registry = serde_json::from_str(json).expect("valid json file");
+    registry.is_valid()?;
+
+    let mut registry = registry.registry;
+    // Sort by name so that we can later binary search by name
+    registry.sort_by_key(|a| a.name());
+
+    // Variables to insert into quote template:
+    let identifier: Vec<_> = registry
+        .iter()
+        .map(|r| format_ident!("{}", r.name()))
+        .collect();
+
+    let reserved_identifiers = registry
+        .iter()
+        .filter(|r| r.is_reserved())
+        .map(|r| format_ident!("{}", r.name()));
+
+    let count = registry.len();
+    let number: Vec<_> = registry.iter().map(|r| r.prefix).collect();
+
+    let mut prefix_to_idx: Vec<_> = number.iter().enumerate().map(|(a, b)| (b, a)).collect();
+    prefix_to_idx.sort_by_key(|(prefix, _)| *prefix);
+    let prefix_to_idx = prefix_to_idx
+        .iter()
+        .map(|(prefix, idx)| quote! { (#prefix, #idx) });
+
+    let name = registry.iter().map(|r| r.network.clone());
+    let desc = registry.iter().map(|r| {
+        if let Some(website) = &r.website {
+            format!("{} - <{}>", r.display_name, website)
+        } else {
+            r.display_name.clone()
+        }
+    });
+
+    Ok(quote! {
+
+        /// A known address (sub)format/network ID for SS58.
+        #[non_exhaustive]
+        #[repr(u16)]
+        #[derive(Copy, Clone, PartialEq, Eq, Debug)]
+        pub enum KnownSs58AddressFormat {
```

Comment:
```suggestion
        pub enum Ss58AddressFormatRegistry {
```

Sounds imo better than `KnownSs58AddressFormat`